opentelemetry-sdk/src/metrics/meter_provider.rs

```diff
@@ -123,15 +123,23 @@ impl MeterProvider for SdkMeterProvider {
         schema_url: Option<impl Into<Cow<'static, str>>>,
         attributes: Option<Vec<KeyValue>>,
     ) -> Meter {
-        let inst_provider: Arc<dyn InstrumentProvider + Send + Sync> =
-            if !self.is_shutdown.load(Ordering::Relaxed) {
-                let scope = Scope::new(name, version, schema_url, attributes);
-                Arc::new(SdkMeter::new(scope, self.pipes.clone()))
-            } else {
-                Arc::new(NoopMeterCore::new())
-            };
+        if self.is_shutdown.load(Ordering::Relaxed) {
+            return Meter::new(Arc::new(NoopMeterCore::new()));
+        }
+
+        let scope = Scope::new(name, version, schema_url, attributes);
 
-        Meter::new(inst_provider)
+        if let Ok(mut meters) = self.meters.lock() {
```

Comment:
Given the fact that obtaining `meter` is not free, we should mention in the documentation that, meter (and instruments) should be obtained once and cached by user to avoid perf hit. Not blocking this PR, but wanted to call it out here.

Reply:
I think we can do some benchmarking in a follow up and have better guidelines on metric perf in general. The use case for accessing a meter is different than a tracer or logger as instruments are more likely to be a fixed set. And even in those cases repeatedly creating/accessing instruments could be the place that needs perf considerations.